plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/expression/ExpressionConverter.java

```diff
@@ -0,0 +1,260 @@
+            if (arguments.isEmpty()) {
+                // Handle case of empty arguments for OR if necessary
+                throw new IllegalArgumentException("OR function expects at least one argument.");
+            }
+            if (arguments.size() == 1) {
+                // OR(X) is just X
+                return arguments.getFirst();
+            }
+            // Iteratively build the OR chain
+            Expression result = arguments.getFirst();
+            for (int i = 1; i < arguments.size(); i++) {
+                result = Predicates.or(result, arguments.get(i));
+            }
+            finalHudiExpression = result;
+        }
+        else if (StandardFunctions.NOT_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            finalHudiExpression = Predicates.not(arguments.getFirst());
+        }
+        // Comparison Operators
+        else if (StandardFunctions.EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.eq(arguments.get(0), arguments.get(1));
+        }
+        else if (StandardFunctions.NOT_EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.not(Predicates.eq(arguments.get(0), arguments.get(1)));
+        }
+        else if (StandardFunctions.GREATER_THAN_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.gt(arguments.get(0), arguments.get(1));
+        }
+        else if (StandardFunctions.LESS_THAN_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.lt(arguments.get(0), arguments.get(1));
+        }
+        else if (StandardFunctions.GREATER_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.gteq(arguments.get(0), arguments.get(1));
+        }
+        else if (StandardFunctions.LESS_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.lteq(arguments.get(0), arguments.get(1));
+        }
+        // Null checks
+        else if (StandardFunctions.IS_NULL_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            finalHudiExpression = Predicates.isNull(arguments.getFirst());
+        }
+        // IN operator from Trino's perspective might be a call like "IN"(column, val1, val2...)
+        // or "IN"(column, ARRAY[val1, val2...]) or it might be handled by TupleDomain primarily.
+        // If it comes as a function:
+        else if (StandardFunctions.IN_PREDICATE_FUNCTION_NAME.equals(functionName) && arguments.size() > 1) {
+            // First argument is the column/expression, rest are values or an array
+            finalHudiExpression = Predicates.in(arguments.getFirst(), arguments.subList(1, arguments.size()));
```

Comment:
In the case of `"IN"(column, ARRAY[val1, val2...])`, does the second argument only contain `ARRAY[val1, val2...]` and Hudi's `Predicates` supports it?

Reply:
Or should `ARRAY[val1, val2...]` be unfolded to adapt to Hudi's `Predicates`?